geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -447,6 +447,59 @@ object SimpleFeatureTypes {
     renamed
   }
 
+  /**
+    * Compare two feature types. This method compares the the schemas of each feature type, i.e. the
+    * names and type bindings of the attributes. The result will be:
+    *
+    * <ul>
+    *   <li>a positive int if type A is a sub type or reorder of type B</li>
+    *   <li>zero if type A has the same schema as type B</li>
+    *   <li>a negative int if none of the above</li>
+    * </ul>
+    *
+    * Note that this shouldn't be used for sorting.
+    *
+    * Note: GeoTools DataUtilities has a similar method that does not work correctly
+    *
+    * @param a schema A
+    * @param b schema B
+    * @return comparison
+    */
+  def compare(a: SimpleFeatureType, b: SimpleFeatureType): Int = {
```

Comment:
Oh man, does this compare function satisfy the contract for Comparator: https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html?

Reply:
no, that's why it says don't use for sorting :)

Reply:
Just like a compiler, I don't read comments.

Unlike a compiler, I only read some of the actual code.

I wonder if there's a better name for this function (so that folks don't blow up in a sort)....

Reply:
I'm not even sure what sorting simple feature types would mean... I copied the name from data utilities for consistency, but I can change it if you want. We're also not using the 'subtype' comparison part, but it seemed like we might as well have it available if needed.